SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -0,0 +1,545 @@
+		return true;
+	}
+	else
+	{
+		UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Spatial service failed to start! %s"), *ServiceStartResult);
+		bSpatialServiceRunning = false;
+		bLocalDeploymentRunning = false;
+		return false;
+	}
+}
+
+bool FLocalDeploymentManager::TryStopSpatialService()
+{
+	if (!bSpatialServiceRunning)
+	{
+		UE_LOG(LogSpatialDeploymentManager, Log, TEXT("Tried to stop spatial service but it's not running."));
+		return false;
+	}
+
+	bStoppingSpatialService = true;
+	FString SpatialExe = TEXT("spatial.exe");
+	FString SpatialServiceStartArgs = TEXT("service stop");
+	const FString SpatialDirectory = FSpatialGDKServicesModule::GetSpatialOSDirectory();
+
+	FString ServiceStopResult;
+	int32 ExitCode;
+	ExecuteAndReadOutput(*SpatialExe, *SpatialServiceStartArgs, *SpatialDirectory, ServiceStopResult, ExitCode);
+	bStoppingSpatialService = false;
+
+	if (ExitCode == ExitCodeSuccess)
+	{
+		UE_LOG(LogSpatialDeploymentManager, Log, TEXT("Spatial service stopped!"));
+		bSpatialServiceRunning = false;
+		bLocalDeploymentRunning = false;
+		return true;
+	}
+	else
+	{
+		UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Spatial service failed to stop! %s"), *ServiceStopResult);
+	}
+
+	IsSpatialServiceRunning();
+	return false;
+}
+
+bool FLocalDeploymentManager::GetLocalDeploymentStatus()
+{
+	if (!bSpatialServiceRunning)
+	{
+		bLocalDeploymentRunning = false;
+		return bLocalDeploymentRunning;
+	}
+
+	FString SpotExe = GetSpotExe();
+	FString SpotListArgs = FString::Printf(TEXT("alpha deployment list --project-name=%s --json --view BASIC --status-filter NOT_STOPPED_DEPLOYMENTS"), *ProjectName);
+	const FString SpatialDirectory = FSpatialGDKServicesModule::GetSpatialOSDirectory();
+
+	FString SpotListResult;
+	FString StdErr;
+	int32 ExitCode;
+	FPlatformProcess::ExecProcess(*SpotExe, *SpotListArgs, &ExitCode, &SpotListResult, &StdErr, *SpatialDirectory);
+
+	if (ExitCode != ExitCodeSuccess)
+	{
+		UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Failed to check local deployment status: %s"), *StdErr);
+		return false;
+	}
+
+	TSharedPtr<FJsonObject> SpotJsonResult;
+	bool bPasingSuccess = ParseJson(SpotListResult, SpotJsonResult);
+	if (!bPasingSuccess)
+	{
+		UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Json parsing of spot list result failed."));
+	}
+
+	const TSharedPtr<FJsonObject>* SpotJsonContent = nullptr;
+	if (bPasingSuccess && SpotJsonResult->TryGetObjectField(TEXT("content"), SpotJsonContent))
+	{
+		const TArray<TSharedPtr<FJsonValue>>* JsonDeployments;
+		if (!SpotJsonContent->Get()->TryGetArrayField(TEXT("deployments"), JsonDeployments))
+		{
+			UE_LOG(LogSpatialDeploymentManager, Verbose, TEXT("No local deployments running."));
+			return false;
+		}
+
+		for (TSharedPtr<FJsonValue> JsonDeployment : *JsonDeployments)
+		{
+			FString DeploymentStatus;
+			if (JsonDeployment->AsObject()->TryGetStringField(TEXT("status"), DeploymentStatus))
+			{
+				if (DeploymentStatus == TEXT("RUNNING"))
```

Comment:
Any documentation regarding the "RUNNING" constant?

Reply:
I wish. RUNNING is just what we can parse out of the output.